### ISSUE TITLE
Paypal response error (TypeError: a bytes-like object is required, not 'str')

### ISSUE THREAD

**Issue by go4site** [odoo/odoo#21156]
Impacted versions: Odoo 11.0

Steps to reproduce: Try to complete payment with Paypal sandbox (Payment data transfer & IPN enabled ). After successful payment during parsing pdt response error is shown "TypeError: a bytes-like object is required, not 'str'"

Current behavior: Not able to parse Paypal response.

Expected behavior: Payment confirmation should be shown. 

Video/Screenshot link (optional):

![paypal_error](https://user-images.githubusercontent.com/5569632/33159715-f1333f48-d026-11e7-9def-994b23a83c2e.png)

File Path addons/payment_paypal/controllers/main.py 
If I change
`lines = [line for line in response.split('\n') if line]`
to
`lines = [line for line in response.split(b'\n') if line]
`
Same error reports for other lines where adding appending  b resolve the error.  After that if I receive 

```
pdt_post[split[0]] = urls.url_unquote_plus(split[1]).decode('utf8')
AttributeError: 'str' object has no attribute 'decode'

```
removing .decode('utf8') at line 48 resolve this error. 

Finally next error appears:

```
/addons/payment_paypal/models/payment.py", line 140, in _paypal_form_get_tx_from_data
    def _paypal_form_get_tx_from_data(self, data):
        reference, txn_id = data.get('item_number'), data.get('txn_id')
        if not reference or not txn_id:
            error_msg = _('Paypal: received data with missing reference (%s) or txn_id (%s)') % (reference, txn_id)
            _logger.info(error_msg)
            raise ValidationError(error_msg)
 
        # find tx -> @TDENOTE use txn_id ?
        txs = self.env['payment.transaction'].search([('reference', '=', reference)])
        if not txs or len(txs) > 1:
            error_msg = 'Paypal: received data for reference %s' % (reference)
odoo.exceptions.ValidationError: ('Paypal: received data with missing reference (None) or txn_id (None)', None)

```
Below is the data received from Paypal:

> odoo.addons.payment_paypal.controllers.main: Beginning Paypal DPN form_feedback with post data {'amt': '20.00',
>  'cc': 'USD',
>  'cm': '{"return_url": "/shop/payment/validate"}',
>  'item_name': 'My Company: SO008',
>  'item_number': 'SO008',
>  'st': 'Completed',
>  'tx': '2CA49582CA7319924'}
> 

PS: I am new to programming world. Please accept apology if this issue report does not provide expected information/workflow.

**Comment by Yenthe666** [comment]
@xmo-odoo this looks like one for you. Another issue with encoding and accessing the data.

**Comment by ilmir-k** [comment]
I have the following error when use Paypal sandbox:
```
File "/mnt/odoo-extra/runbot/static/build/11210-236-d5c40d/odoo/addons/payment_paypal/controllers/main.py", line 41, in _parse_pdt_response
    lines = [line for line in response.split('\n') if line]
TypeError: Type str doesn't support the buffer API
```

I get it at final step after successful payment and redirecting to return url

**Comment by xmo-odoo** [comment]
Should be fixed in latest 11.0, check if it now works properly.

**Comment by go4site** [comment]
Thanks a lot. I will check and confirm asap. Thanks again!
